File: src/main/java/com/example/demo/api/ErrorMessage.java
package com.example.demo.api;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ErrorMessage {
	private final String code;
    private final String message;
    
    public static ErrorMessage from(ErrorCode errorCode) {
    	return new ErrorMessage(errorCode.code(), errorCode.message());
    }
    public static ErrorMessage from(ErrorCode errorCode, String customErrorMessage) {
    	return new ErrorMessage(errorCode.code(), customErrorMessage);
    }
    
}

File: src/main/kotlin/com/example/demo/api/ErrorMessage.kt
package com.example.demo.api

data class ErrorMessage(
    val code: String,
    val message: String
) {
    companion object {
        fun from(errorCode: ErrorCode, customErrorMessage: String? = null): ErrorMessage {
            return ErrorMessage(
                code = errorCode.code,
                message = customErrorMessage ?: errorCode.message
            )
        }
    }
}
